src/player/Marine.cpp

```diff
@@ -82,6 +82,101 @@ int32_t Marine::checkForPickUp() {
     return -1;
 }
 
+/**
+* Date: Mar 27 
+* Author: Aing Ragunathan
+*
+* Interface: void Marine::updateImageDirection()
+*
+* Description:
+*       This function changes the direction that the character is facing.
+*       It is called from GameStateMatch::handle after every frame and
+*       updates the direction of the sprite according to the angle of the
+*       mouse from the center of the screen.
+*/
+void Marine::updateImageDirection() {
+    const double radians = (getAngle() -90) * M_PI/180;
+
+    //order: start from ~0 rad, counter clockwise
+    if (radians > SPRITE_ANGLE2 && radians < SPRITE_ANGLE1) { 
+        setSrcRect(getSrcRect().x, SPRITE_RIGHT, SPRITE_SIZE_X, SPRITE_SIZE_Y);
+    } else if (radians > SPRITE_ANGLE3 && radians < SPRITE_ANGLE2) { 
+        setSrcRect(getSrcRect().x, SPRITE_BACK_RIGHT, SPRITE_SIZE_X, SPRITE_SIZE_Y); 
+    } else if (radians > SPRITE_ANGLE4 && radians < SPRITE_ANGLE3) { 
+        setSrcRect(getSrcRect().x, SPRITE_BACK, SPRITE_SIZE_X, SPRITE_SIZE_Y); 
+    } else if (radians > SPRITE_ANGLE5 && radians < SPRITE_ANGLE4) { 
+        setSrcRect(getSrcRect().x, SPRITE_BACK_LEFT, SPRITE_SIZE_X, SPRITE_SIZE_Y); 
+    } else if (radians > SPRITE_ANGLE6 && radians < SPRITE_ANGLE5) { 
+        setSrcRect(getSrcRect().x, SPRITE_LEFT, SPRITE_SIZE_X, SPRITE_SIZE_Y); 
+    } else if (radians > SPRITE_ANGLE7 && radians < SPRITE_ANGLE6) { 
+        setSrcRect(getSrcRect().x, SPRITE_FRONT_LEFT, SPRITE_SIZE_X, SPRITE_SIZE_Y); 
+    } else if (radians < SPRITE_ANGLE7) { 
+        setSrcRect(getSrcRect().x, SPRITE_FRONT, SPRITE_SIZE_X, SPRITE_SIZE_Y); 
+    } else if (radians > SPRITE_ANGLE1 && radians < SPRITE_ANGLE8) { 
+        setSrcRect(getSrcRect().x, SPRITE_FRONT_RIGHT, SPRITE_SIZE_X, SPRITE_SIZE_Y);
+    }    
+}
+
+/**
+* Date: Mar 30 
+* Author: Aing Ragunathan
+*
+* Function Interface: void Marine::updateImageWalk(double frameCount)
+*       double frameCount - counted frames while walking
+*
+* Description:
+*       This function repeatedly updates the image of the marine in order to animate 
+*       walking. It is called from GameStateMatch::handle after every frame and updates
+*       the feet placement of the marine.
+*/
+void Marine::updateImageWalk(const Uint8 *state, double frameCount) {
```

Comment:
const also frame should be the same type as the getTicks return as that should be what its receiving